sdk/core/azure-core-amqp/src/models/amqp_value.cpp

```diff
@@ -128,7 +128,7 @@ namespace Azure { namespace Core { namespace Amqp { namespace Models {
     char value;
     if (amqpvalue_get_byte(m_value.get(), &value) != 0)
     {
-      throw std::runtime_error("Could not retrieve value");
+      throw std::runtime_error("Could not retrieve byte value");
```

Comment:
Do we need both int8_t and char? If so, consider matching the type name in the exception message rather than exposing an implementation detail that the underlying method called is `get_byte(...)` in both cases.
```suggestion
      throw std::runtime_error("Could not retrieve char value");
```

Reply:
The 'char' overload is a convenience overload to allow someone to write `AmqpValue{'Z'}`. It's analogous to the `char *` override for strings.

The underlying AQMP type is `byte`, and the error message correctly reflects that.